File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionContains.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.*;

import java.util.Map;

public class TransformerFunctionContains extends TransformerFunction {
    public TransformerFunctionContains() {
        super(FunctionDescription.of(
            Map.of(
            "that", ArgumentType.of(ArgType.Any).position(0).defaultIsNull(true)
            )
        ));
    }
    @Override
    public CompletionStage<Object> apply(FunctionContext context) {
        var streamer = context.getJsonElementStreamer(null);
        if (streamer == null) return null;
        var that = context.getJsonElement( "that");
        return streamer.stream().anyMatch(el -> el.equals(that));
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionContains.ts
import TransformerFunction from "./common/TransformerFunction";
import { ArgType } from "./common/ArgType";
import FunctionContext from "./common/FunctionContext";
import { isEqual } from "../JsonHelpers";

class TransformerFunctionContains extends TransformerFunction {
  constructor() {
    super({
      arguments: {
        that: { type: ArgType.Any, position: 0, defaultIsNull: true },
      },
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const streamer = await context.getJsonElementStreamer(null);
    if (streamer == null) return null;
    const that = await context.getJsonElement("that");
    return streamer.stream().any(el => isEqual(el, that));
  }
}

export default TransformerFunctionContains;
